parameters.cpp

```diff
@@ -553,67 +554,49 @@ void Parameters::getParameters() {
    RP::get("AMR.refine_limit", P::amrRefineLimit);
    RP::get("AMR.coarsen_limit", P::amrCoarsenLimit);
    RP::get("AMR.transShortPencils", P::amrTransShortPencils);
-
-   /*Read Blur Passes per Refinement Level*/
    RP::get("AMR.filterpasses", P::blurPassString);
 
-   // Construct Vector of Passes used in grid.cpp
-   bool isEmpty = blurPassString.size() == 0;
-   vector<int>::iterator maxNumPassesPtr;
-   int maxNumPassesInt;
-
-   if (!isEmpty) {
-
-      for (auto i : blurPassString) {
-         P::numPasses.push_back(stoi(i));
-      }
-
-      // Reverse Sort and Get the maximum number of filter passes
-      sort(numPasses.begin(), numPasses.end(), greater<int>());
-
-      // Sanity Check
-      if (P::numPasses.size() != P::amrMaxSpatialRefLevel + 1) {
-         cerr << "Filter Passes=" << P::numPasses.size() << "\t"
-              << "AMR Levels=" << P::amrMaxSpatialRefLevel + 1 << endl;
-         cerr << "FilterPasses do not match AMR levels \t"
-              << " in " << __FILE__ << ":" << __LINE__ << endl;
-         MPI_Abort(MPI_COMM_WORLD, 1);
-      }
-
-      if (myRank == MASTER_RANK) {
-
-         maxNumPassesPtr = max_element(P::numPasses.begin(), P::numPasses.end());
-         if (maxNumPassesPtr != numPasses.end()) {
-            maxNumPassesInt = *maxNumPassesPtr;
-         } else {
-            cerr << "Trying to dereference null pointer \t"
-                 << " in " << __FILE__ << ":" << __LINE__ << endl;
+   // If we are in an AMR run we need to set up the filtering scheme.
+   if (P::amrMaxSpatialRefLevel>0){
+      bool isEmpty = blurPassString.size() == 0;
+      if (!isEmpty){
+         //sanity check=> user should define a pass for every level
+         if (blurPassString.size() != P::amrMaxSpatialRefLevel + 1) {
+            cerr << "Filter Passes=" << blurPassString.size() << "\t" << "AMR Levels=" << P::amrMaxSpatialRefLevel + 1 << endl;
+            cerr << "FilterPasses do not match AMR levels. \t" << " in " << __FILE__ << ":" << __LINE__ << endl;
             MPI_Abort(MPI_COMM_WORLD, 1);
          }
-
-         printf("Filtering is on with max number of Passes= \t%d\n", maxNumPassesInt);
-         int lev = 0;
-         for (auto& iter : P::numPasses) {
-            printf("Refinement Level %d-->%d Passes\n", lev, iter);
-            lev++;
+         //sort the filtering passes per refLevel
+         numPasses.clear();
+         std::sort(blurPassString.begin(), blurPassString.end(),
+                  [](const std::string& lhs, const std::string& rhs)->bool 
+                  { return (lhs.size() < rhs.size()) || (lhs.size() == rhs.size() && lhs < rhs);});
```

Comment:
why are you sorting based on the size of a string? Shouldn't you just convert the string contents to a vector of ints and sort that?